From leitao@linux.vnet.ibm.com  Fri Feb 26 16:30:15 2010
From: Breno Leitao <leitao@linux.vnet.ibm.com>
Date: Thu, 25 Feb 2010 15:31:49 -0300
Subject: jsm: removing the uart structure and filename on error
To: greg@kroah.com
Message-ID: <daf3e60a90070795196e941ac058335ead36ce14.1267122632.git.leitao@linux.vnet.ibm.com>


If jsm fails to load, then remove the uart stuff, otherwise,
the things (as files), will be there forever (even when the module
is unloaded). If you try to reload the module, the following message
appears:

kobject_add_internal failed for ttyn1 with -EEXIST, don't try to
register things with the same name in the same directory.

This patch remove the uart things when the driver fails.

Signed-off-by: Breno Leitao <leitao@linux.vnet.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/jsm/jsm_driver.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/serial/jsm/jsm_driver.c
+++ b/drivers/serial/jsm/jsm_driver.c
@@ -179,6 +179,7 @@ static int __devinit jsm_probe_one(struc
 
 	return 0;
  out_free_irq:
+	jsm_remove_uart_port(brd);
 	free_irq(brd->irq, brd);
  out_iounmap:
 	iounmap(brd->re_map_membase);
